src/patternfly/components/CodeEditor/code-editor.scss

```diff
@@ -70,62 +87,63 @@
   }
 }
 
-.#{$code-editor}__controls {
+.#{$code-editor}__header-content {
   display: flex;
+  flex: 1;
+  align-items: flex-end;
+  padding-block-start: var(--#{$code-editor}__header-content--PaddingBlockStart);
+  padding-block-end: var(--#{$code-editor}__header-content--PaddingBlockEnd);
+  background-color: var(--#{$code-editor}__header--BackgroundColor);
+  border-radius: var(--#{$code-editor}__tab--BorderRadius) 0 0 0;
 
-  .#{$button}.pf-m-control {
-    --#{$button}--m-control--Color: var(--#{$code-editor}__controls--c-button--m-control--Color);
-
-    &:hover {
-      --#{$code-editor}__controls--c-button--m-control--Color: var(--#{$code-editor}__controls--c-button--m-control--hover--Color);
-    }
-
-    &:focus {
-      --#{$code-editor}__controls--c-button--m-control--Color: var(--#{$code-editor}__controls--c-button--m-control--focus--Color);
-    }
-
-    &:disabled {
-      &::after {
-        border-block-end-color: var(--#{$code-editor}__controls--c-button--m-control--disabled--after--BorderBottomColor);
-      }
-    }
+  &.pf-m-plain {
+    --#{$code-editor}__header--BackgroundColor: var( --#{$code-editor}__header--m-plain--BackgroundColor);
   }
 }
 
+.#{$code-editor}__controls {
```

Comment:
Might be worth adding flex-wrap to this, for when  the amount of control buttons causes the container to shove everything off screen:

![image](https://github.com/patternfly/patternfly/assets/70952936/9b366eee-6cec-4b68-8d99-ca73ca327d73)